src/Confluent.Kafka/IProducer.cs

```diff
@@ -0,0 +1,150 @@
+        ISerializingProducer<TKey, TValue> GetSerializingProducer<TKey, TValue>(ISerializer<TKey> keySerializer, ISerializer<TValue> valueSerializer);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_Message"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_Common"]/*' />
+        Task<Message> ProduceAsync(Message message);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_string_TKey_TValue"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_Common"]/*' />
+        Task<Message> ProduceAsync(string topic, byte[] key, byte[] val);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_string_Partition_TKey_TValue_Timestamp_IEnumerable"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_Common"]/*' />
+        Task<Message> ProduceAsync(
+            string topic, Partition partition, 
+            byte[] key, byte[] val, 
+            Timestamp timestamp, 
+            IEnumerable<KeyValuePair<string, byte[]>> headers);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_string_Partition_byte_int_int_byte_int_int_Timestamp_IEnumerable"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_Common"]/*' />
+        Task<Message> ProduceAsync(
+            string topic, Partition partition, 
+            byte[] key, int keyOffset, int keyLength, 
+            byte[] val, int valOffset, int valLength, 
+            Timestamp timestamp, IEnumerable<KeyValuePair<string, byte[]>> headers);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_Message"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="Produce_IDeliveryHandler"]/*' />
+        void Produce(Message message, IDeliveryHandler deliveryHandler);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_string_TKey_TValue"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="Produce_IDeliveryHandler"]/*' />
+        void Produce(string topic, byte[] key, byte[] val, IDeliveryHandler deliveryHandler);
+
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="ProduceAsync_string_Partition_TKey_TValue_Timestamp_IEnumerable"]/*' />
+        /// <include file='include_docs_producer.xml' path='API/Member[@name="Produce_IDeliveryHandler"]/*' />
+        void Produce(
+            string topic, Partition partition,
+            byte[] key, byte[] val,
+            Timestamp timestamp,
+            IEnumerable<KeyValuePair<string, byte[]>> headers,
+            IDeliveryHandler deliveryHandler);
```

Comment:
going to replace `IDeliveryHandler` with `IAction<Message>` since using `IDeliveryHandler` is a bit awkward to use and in practice provides little benefit.

Reply:
going to defer doing this to a latter PR to cut scope in this one, but want to get it in before the next release to get all the API changes done at once.

Reply:
yes, but I think it's justifiable to do it whilst we're making these other changes. anyway, doesn't affect this PR.